Library/Homebrew/tap.rb

```diff
@@ -1279,7 +1281,7 @@ def cask_tokens
   def cask_files_by_name
     return super if Homebrew::EnvConfig.no_install_from_api?
 
-    Homebrew::API::Cask.all_casks.each_with_object({}) do |item, hash|
+    @cask_files_by_name ||= Homebrew::API::Cask.all_casks.each_with_object({}) do |item, hash|
```

Comment:
Hmm, we probably should avoid Pathname here too

Reply:
Agreed.